src/api/js/src/high-level/types.ts

```diff
@@ -953,11 +1197,110 @@ export interface BitVecNum<Bits extends number = number, Name extends string = '
   readonly __typename: 'BitVecNum';
 
   value(): bigint;
+
   asSignedValue(): bigint;
+
   asString(): string;
+
   asBinaryString(): string;
 }
 
+/**
+ * A Sort representing a SMT Array with range of sort {@link SMTArraySort.range range}
+ * and a domain of sort {@link SMTArraySort.domain domain}
+ *
+ * @typeParam RangeSort The sort of the array range
+ * @category Arrays
+ */
+export interface SMTArraySort<DomainSort extends AnySort<Name>[],
+  RangeSort extends AnySort<Name>,
+  Name extends string = 'main'> extends Sort<Name> {
+  /** @hidden */
+  readonly __typename: 'ArraySort';
+
+  /**
+   * The sort of the first dimension of the domain
+   *
+   * TODO: Add example
+   */
+  domain(): DomainSort[0];
+
+  /**
+   * The sort of the i-th (0-indexed) dimension of the domain
+   *
+   * TODO: add example
+   *
+   * @param i index of the dimension of the domain being requested
+   */
+  domain_n(i: number): DomainSort[number];
```

Comment:
I don't know if TypeScript will let you get away with it, but if it will,

```suggestion
  domain_n<T extends number>(i: T): DomainSort[T];
```

That way when you pass a constant value, like `.domain_n(2)`, type inference can get the actual domain.

Reply:
Nice!